compiler/plugins/target/ROCM/builtins/tuning/iree_default_tuning_spec_gfx942.mlir

```diff
@@ -19,6 +19,77 @@ transform.named_sequence @apply_op_config(%op: !transform.any_op {transform.read
+  transform.yield
+}
+
+transform.named_sequence @match_attention_f16(%root: !transform.any_op {transform.readonly})
+  -> !transform.any_op {
+  transform.match.operation_name %root ["iree_linalg_ext.attention"] : !transform.any_op
+  %ins, %outs = transform.iree.match.cast_compatible_dag_from_root %root {
+    ^bb0(%query: tensor<?x?x?x?xf16>,
+         %key: tensor<?x?x?x?xf16>,
+         %value: tensor<?x?x?x?xf16>,
+         %softmax_scale: f16,
+         %out: tensor<?x?x?x?xf16>):
+
+      %attn = iree_linalg_ext.attention {indexing_maps = [
+                                          affine_map<(d0, d1, d2, d3, d4, d5) -> (d0, d1, d2, d4)>,
+                                          affine_map<(d0, d1, d2, d3, d4, d5) -> (d0, d1, d5, d4)>,
+                                          affine_map<(d0, d1, d2, d3, d4, d5) -> (d0, d1, d3, d5)>,
+                                          affine_map<(d0, d1, d2, d3, d4, d5) -> ()>,
+                                          affine_map<(d0, d1, d2, d3, d4, d5) -> (d0, d1, d2, d3)>]}
+        ins(%query, %key, %value, %softmax_scale :
+            tensor<?x?x?x?xf16>, tensor<?x?x?x?xf16>, tensor<?x?x?x?xf16>, f16)
+        outs(%out : tensor<?x?x?x?xf16>){
+          ^bb0(%arg0: f32):
+            iree_linalg_ext.yield %arg0 : f32
+        } -> tensor<?x?x?x?xf16>
+  } : (!transform.any_op) -> (!transform.any_value, !transform.any_value)
+
+  transform.yield %root : !transform.any_op
+}
+
+transform.named_sequence
+@match_attention_2x10x4096x64x64x64_f16(%attention: !transform.any_op {transform.readonly})
+  -> (!transform.any_op, !transform.any_param, !transform.any_param) {
+
+  %matched = transform.include @match_attention_f16 failures(propagate) (%attention)
+    : (!transform.any_op) -> !transform.any_op
+
+  %query = transform.get_operand %attention[0] : (!transform.any_op) -> !transform.any_value
+  %key = transform.get_operand %attention[1] : (!transform.any_op) -> !transform.any_value
+  %value = transform.get_operand %attention[2] : (!transform.any_op) -> !transform.any_value
+
+  transform.iree.match.cast_compatible_type %query = tensor<2x10x4096x64xf16> : !transform.any_value
+  transform.iree.match.cast_compatible_type %key = tensor<2x10x64x64xf16> : !transform.any_value
+  transform.iree.match.cast_compatible_type %value = tensor<2x10x64x64xf16> : !transform.any_value
+
+  %config = transform.param.constant #iree_codegen.compilation_info<
+          lowering_config = #iree_gpu.lowering_config<{workgroup = [1, 1, 128, 0, 0, 0], reduction=[0, 0, 0, 0, 0, 64], promote_operands = [1, 2]}>,
+          translation_info = #iree_codegen.translation_info<pipeline = LLVMGPUVectorDistribute
+                                                            workgroup_size = [64, 4]
+                                                            subgroup_size = 64 ,
+            {llvm_func_attrs = { "amdgpu-waves-per-eu" = "2", "denormal-fp-math-f32" = "preserve-sign" }}>>
+  -> !transform.any_param
```

Comment:
amdgpu-waves-per-eu reasoning:

Our attention implementation on MI300X uses a large number of registers. Setting this flag makes the compiler less conservative when assigning registers. (@kuhar may know a better reasoning actually).

denormal-fp-math-f32:

This asks the llvm backend to disable denorm flushing for exp2/exp, which consumes less instructions on doing exp/exp2.